build-system/release-workflows/promote-nightly.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const {log} = require('../common/logging');
+const {Octokit} = require('@octokit/rest');
+const {runReleaseJob} = require('./release-job');
+const {VERSION} = require('../compile/internal-version');
+
+/**
+ * @fileoverview Script that promotes the latest nightly release.
+ */
+
+const jobName = 'promote-nightly.js';
+
+const octokit = new Octokit({auth: process.env.GITHUB_TOKEN});
+
+const versioningJsonFile = 'build-system/global-configs/versioning.json';
+const params = {owner: 'ampproject', repo: 'amphtml'};
+
+runReleaseJob(jobName, async () => {
+  const getContentResponse = await octokit.repos.getContent({
+    ...params,
+    path: versioningJsonFile,
+    ref: 'main',
+  });
+  if (!('content' in getContentResponse.data)) {
+    throw new Error(`Failed to fetch ${versioningJsonFile}`);
+  }
+
+  const versioning = Object.assign(
+    JSON.parse(
+      Buffer.from(getContentResponse.data.content, 'base64').toString('utf8')
+    ),
+    {nightly: `04${VERSION}`}
+  );
+
+  const updateFileResponse = await octokit.repos.createOrUpdateFileContents({
```

Comment:
I kind of find it difficult to believe that this will work given our branch protection rules. @rsimha do you have an idea how to give one job the permission to directly commit to the version change?

Alternatively we can create a PR, enable auto-merge on it, and have some other automated process approve that PR so it'll be merged. WDYT?

Reply:
If it's possible to ensure that the github action is run by a well-known bot account like `ampprojectbot`, we can work around the _"Restrict who can push to matching branches"_ setting by adding our bot to the allowlist for `main`.

> Alternatively we can create a PR, enable auto-merge on it, and have some other automated process approve that PR so it'll be merged. WDYT?

After that, what you suggest here should work.